src/Core/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -1140,21 +1140,38 @@ public static bool IsGraphQLReservedName(Entity entity, string databaseColumnNam
             };
 
             string tablePrefix = GetTablePrefix(conn.Database, schemaName);
+            string queryPrefix = string.IsNullOrEmpty(tablePrefix) ? string.Empty : $"{tablePrefix}.";
             selectCommand.CommandText
-                = $"SELECT * FROM {tablePrefix}.{SqlQueryBuilder.QuoteIdentifier(tableName)}";
+                = $"SELECT * FROM {queryPrefix}{SqlQueryBuilder.QuoteIdentifier(tableName)}";
             adapterForTable.SelectCommand = selectCommand;
 
             DataTable[] dataTable = adapterForTable.FillSchema(EntitiesDataSet, SchemaType.Source, tableName);
             return dataTable[0];
         }
 
-        private string GetTablePrefix(string databaseName, string schemaName)
+        internal string GetTablePrefix(string databaseName, string schemaName)
         {
-            StringBuilder tablePrefix = new(SqlQueryBuilder.QuoteIdentifier(databaseName));
-            if (!string.IsNullOrEmpty(schemaName))
+            StringBuilder tablePrefix = new();
+
+            if (!string.IsNullOrEmpty(databaseName))
+            {
+                // Determine databaseName for prefix.
+                databaseName = SqlQueryBuilder.QuoteIdentifier(databaseName);
+                tablePrefix.Append(databaseName);
+
+                if (!string.IsNullOrEmpty(schemaName))
+                {
+                    // Determine schemaName for prefix.
+                    schemaName = SqlQueryBuilder.QuoteIdentifier(schemaName);
+                    tablePrefix.Append($".{schemaName}");
+                }
+            }
+            else if (!string.IsNullOrEmpty(schemaName))
             {
+                // Determine schemaName for prefix.
                 schemaName = SqlQueryBuilder.QuoteIdentifier(schemaName);
-                tablePrefix.Append($".{schemaName}");
+                // Database name is empty we just need the schema name.
+                tablePrefix.Append(schemaName);
```

Comment:
any way to simplify this code since it's slightly duplicated except for adding the period?

Reply:
I could create a seperate function that does the tablePrefix append -> code on lines 1172 and 1165 but it seems easier to understand this way.

Reply:
Will resolve based on offline discussion.